### ISSUE TITLE
内存泄漏？

### ISSUE THREAD

**Issue by Feynman83** [qicosmos/cinatra#144]
我在一个只有64MB内存的嵌入式机器上跑例子上的代码，
用AB 做压测，内存一直增长

**Comment by qicosmos** [comment]
是哪个测试代码？把代码贴上来。

**Comment by Feynman83** [comment]
WebAPI::WebAPI() {
  ConfigSystem& cfgSys = ConfigSystem::GetInstance();
  server = new http_server(4);

  bool r = server->listen("0.0.0.0", cfgSys.GetAPIPort());
  if (!r) {
    return;
  }

  server->enable_http_cache(false);  // set global cache
  server->set_res_cache_max_age(86400);
  server->set_cache_max_age(5);
  server->set_http_handler<GET, POST>("/", [](request& req, response& res) {
    json js;
    js["code"] = 0;
    res.render_json(js);
  });

  new ModemHandler(server);
  std::thread tServer([&] { server->run(); });

  tServer.detach();
}

**Comment by Feynman83** [comment]
设备是新唐的ARM9 内置的64MB DDR flash，单核，没有硬件浮点单元

**Comment by qicosmos** [comment]
server不要用指针，用对象，看你这个server貌似都没有删除。

**Comment by qicosmos** [comment]
应该是你使用裸指针没有删除的原因吧。modern c++中永远不要用裸指针，用智能指针。

**Comment by Feynman83** [comment]
我使用postman的测试10000次倒没发现内存增长。
